crates/ruff_python_formatter/src/snapshots/ruff_python_formatter__tests__ruff_test__statement__for_py.snap

```diff
@@ -0,0 +1,87 @@
+---
+source: crates/ruff_python_formatter/src/lib.rs
+expression: snapshot
+---
+## Input
+```py
+for x in y: # trailing test comment
+    pass # trailing last statement comment
+
+    # trailing for body comment
+
+# leading else comment
+
+else: # trailing else comment
+    pass
+
+    # trailing else body comment
+
+
+for aVeryLongNameThatSpillsOverToTheNextLineBecauseItIsExtremelyLongAndGoesOnAndOnAndOnAndOnAndOnAndOnAndOnAndOnAndOn in anotherVeryLongNameThatSpillsOverToTheNextLineBecauseItIsExtremelyLongAndGoesOnAndOnAndOnAndOnAndOnAndOnAndOnAndOnAndOn: # trailing comment
+    pass
+
+else:
+    ...
+
+for (
+    x,
+    y,
+    ) in z: # comment
+    ...
+
+
+# remove brackets around x,y but keep them around z,w
+for (x, y) in (z, w):
+    ...
+
+
+# type comment
+for x in (): # type: int
+    ...
+```
+
+
+
+## Output
+```py
+for x in y:  # trailing test comment
+    pass  # trailing last statement comment
+
+    # trailing for body comment
+
+# leading else comment
+
+else:  # trailing else comment
+    pass
+
+    # trailing else body comment
+
+
+for (
```

Comment:
black doesn't break this for statement, but i think this actually makes more sense

Reply:
hm. interesting (how did you notice the difference)

at this point, do we have a policy/guidelines for matching black (to help people migrate) vs improving formatting? maybe we raise an issue or otherwise to see if the black maintainers also prefer this formatting?

Reply:
Our goal is to match black except in edge cases, but we haven't formalized what exactly this means. I'll discuss with @MichaReiser tomorrow what we want do about this

> how did you notice the difference?

copied the code and ran black over it ;)

Reply:
at the moment, yes